File: src/main/java/com/qtech/bubbles/common/entity/ModifierType.java
package com.qtech.bubbles.common.entity;

import java.util.HashMap;

public class ModifierType {
    static final HashMap<String, ModifierType> types = new HashMap<>();

    private final String name;

    public ModifierType(String name) {
        if (types.containsKey(name)) {
            throw new IllegalArgumentException("Duplicate modifier detected!");
        }

        ModifierType.types.put(name, this);
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: src/main/kotlin/qtech/bubbles/common/entity/ModifierType.kt
package qtech.bubbles.common.entity

class ModifierType(name: String) {
    val name: String

    companion object {
        val types = HashMap<String, ModifierType>()
    }

    init {
        require(!types.containsKey(name)) { "Duplicate modifier detected!" }
        types[name] = this
        this.name = name
    }
}